tensorflow/python/kernel_tests/matmul_op_test.py

```diff
@@ -52,7 +53,12 @@ def _AddTest(test, op_name, testcase_name, fn):
   test_name = "_".join(["test", op_name, testcase_name])
   if hasattr(test, test_name):
     raise RuntimeError("Test %s defined more than once" % test_name)
-  setattr(test, test_name, test_util.deprecated_graph_mode_only(fn))
+  if 'bfloat16' in testcase_name:
+    # running bfloat16 in eager mode as we need to calculate intermediate
+    # tensors by using cast operators
+    setattr(test, test_name, fn)
```

Comment:
I think this won't be necessary once we use `.as_numpy_dtype`.

Reply:
Unfortunately, this is again necessary as while using compute_gradient(fn,x,delta) function, we can't send x as bfloat16, it is giving us wrong numerical values (mainly zero values). So we need to send x as float32.  [more details in the response to comments below]. Using the decorator can't allow to use float32 for x.